### PR TITLE
Load Datasworn JSON files from homebrew folder

### LINKED ISSUE #40: rulesets: add new rulesets using datasworn json files
Allow adding arbitrary rulesets using complete datasworn-schema json (or corresponding yaml) files.

### PR DESCRIPTION
**Pull request by cwegrzyn**
Fixes #40

### CODE REVIEW

File: src/datastore.ts [hunk 1]
```diff
@@ -58,17 +64,35 @@ export class Datastore extends Component implements IDataContext {
...
+        const homebrewFolder = this.plugin.app.vault.getFolderByPath(
+          this.plugin.settings.homebrewPath,
+        );
+        if (homebrewFolder) {
+          await this.indexDataswornFiles(homebrewFolder);
+        } else {
+          new Notice(
+            `Homebrew enabled, but path '${this.plugin.settings.homebrewPath}' is missing.`,
+          );
+          console.warn(
```
Comment (zkat) [inline src/datastore.ts:89]: probably want to use our logger here too.

File: src/datastore.ts [hunk 2]
```diff
@@ -106,6 +130,41 @@ export class Datastore extends Component implements IDataContext {
...
+    const ajv = new Ajv({ strict: false, validateFormats: false });
+    const validate = ajv.compile(dataswornSchema);
+
+    for (const file of folder.children) {
+      if (file instanceof TFile && file.name.endsWith(".json")) {
+        try {
+          const data = JSON.parse(await this.app.vault.cachedRead(file));
+          const result = validate(data);
+          if (!result) {
+            console.error(validate.errors);
```
Comment (zkat) [inline src/datastore.ts:143]: here too

**Review by zkat — APPROVED** [review]
Awesome! This is really exciting! Just pointed out some straggler console calls, but otherwise look good.

Do you want to tackle docs with this PR, or leave them for later?

**Comment by cwegrzyn** [comment]
> Do you want to tackle docs with this PR, or leave them for later?

Ah, good call. Added here!

**Pull request merged**